Gems/PhysX/Code/Include/PhysX/EditorColliderComponentRequestBus.h

```diff
@@ -217,9 +217,9 @@ namespace PhysX
         virtual void ValidateRigidBodyMeshGeometryType() = 0;
     };
 
-    AZ_PUSH_DISABLE_WARNING(4996, "-Wdeprecated-declarations")
+    AZ_PUSH_DISABLE_WARNING(4996, "-Wdeprecated-declarations", "-Wdeprecated-declarations");
```

Comment:
This is a weird change but I assume correct?

Reply:
The second argument i think is for gcc compilation

Reply:
Yes, the second argument only applies to Clang.
The third argument is needed to apply the warning suppression to GCC.

Also the clang warning suppression arguments do **NOT** always map to GCC warning suppression arguments.

While mapping `-Wdeprecated-declarations` to both Clang and GCC works, it doesn't work for other arguments such as `-Wunknown-warning-option`.

So the 2 argument `AZ_PUSH_DISABLE_WARNING` macro cannot be extended to work for both Clang and GCC transparently.